runtime/frequency/src/lib.rs

```diff
@@ -130,16 +130,14 @@ impl Contains<RuntimeCall> for BaseCallFilter {
 		#[cfg(not(feature = "frequency"))]
 		{
 			match call {
-				// Utility Calls are blocked. Issue #599
-				RuntimeCall::Utility(..) => false,
+				RuntimeCall::Utility(..) => Self::is_utility_call_allowed(call),
```

Comment:
I wonder we could do something like the following:
```
impl Contains<RuntimeCall> for BaseCallFilter {
	fn contains(call: &RuntimeCall) -> bool {
		#[cfg(not(feature = "frequency"))]
		{
			match call {
				RuntimeCall::Utility(utility_call) => Self::is_utility_call_allowed(utility_call),
				_ => true,
			}
		}
...

```
impl BaseCallFilter {
	fn is_utility_call_allowed(call: &pallet_utility::Call::<Runtime>) -> bool {
		match call {
			pallet_utility::Call::batch { calls, .. } |
			pallet_utility::Call::batch_all { calls, .. } |
			pallet_utility::Call::force_batch { calls, .. } =>
				calls.iter().any(Self::is_pays_no_call),
			_ => false,
		}
	}
```

Since  RuntimeCall::Utility(......THIS IS THE CALL FROM UTILITY......), we can pass that into `is_utility_call_allowed`.

It seems we are doing it now is passing the RuntimeCall down and making an additional match statement to filter out utility calls. I think this might be able to allow to remove that additional function.

Reply:
I like it, did some parts of it as we still have to check for various `RuntimeCall` along with pays not so the `is_batch_call_allowed` checks for blocked `RuntimeCalls` calls within batches hence this function would require  generic +match

but good suggestion